contracts/token/ERC20/extensions/draft-ERC20Wrapper.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../ERC20.sol";
+import "../utils/SafeERC20.sol";
+
+abstract contract ERC20Wrapper is ERC20 {
+    IERC20 immutable public underlying;
+
+    constructor(IERC20 underlyingToken) {
+        underlying = underlyingToken;
+    }
+
+    function depositFor(address account, uint256 amount) public virtual returns (bool) {
```

Comment:
What do you think of adding a `depositByPermit` function for when the underling token supports ERC20Permit? Although this would add to bytecode size unnecessarily if the underlying token doesn't support it.

Reply:
That is a possibility. I was also considering being a 1363 receiver, so the deposit can be triggered by erc1363 transfer methods

Reply:
How do we deal with all of these variants? It wouldn't be right to include them if they aren't going to be used, but it doesn't feel worth it to make several variants using inheritance...

Reply:
We shouldn't put everything in. I'm not sure what to chose. We should discuss that.

Reply:
Would you want something like this
```
    function depositByPermit(address owner, uint256 value, uint256 deadline, uint8 v, bytes32 r, bytes32 s) public virtual returns (bool) {
        IERC20Permit(underlying).permit(owner, address(this), value, deadline, v, r, s);
        SafeERC20.safeTransferFrom(underlying, owner, address(this), value);
        _mint(owner, value);
        return true;
    }
```
Or would you assume the permit was done in a separate tx, and have a signature that protects that would allow the recipient of the wrapped token to be different then the owner of the underlying asset ?